lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -126,23 +121,35 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
-        return db2_last_insert_id($this->conn);
+        if ($name !== null) {
+            // DB2 does not support sequences. However, PDO::lastInsertId() ignores the name parameter, and returns
+            // the last insert ID even if a sequence name is given. We expect an exception in that case.
+            throw new DB2Exception('DB2 does not support sequences.');
```

Comment:
This may be misleading since IBM DB2 LUW [supports sequences](https://www.ibm.com/support/knowledgecenter/en/SSEPGG_9.7.0/com.ibm.db2.luw.sql.ref.doc/doc/r0004201.html):
> The `CREATE SEQUENCE` statement defines a sequence at the application server.

Reply:
Damn. I checked `DB2Platform::supportsSequences()` and it returns `false`. Should this be fixed?
Anyway, I updated the comment and exception message to remove any confusion.